tensorflow/python/eager/function.py

```diff
@@ -2525,6 +2535,31 @@ def _convert_variables_to_tensors(self, args, kwargs):
     kwargs = {kw: ops.convert_to_tensor(x) for kw, x in kwargs.items()}
     return tuple(args), kwargs
 
+  def _convert_typed_variables_to_tensors(self, args, kwargs):
+    if self.input_signature is not None:
+      return
+
+    args = list(args)
+    for i, arg in enumerate(args):
+      if i < len(self._fullargspec.args):
+        arg_annotation = self._fullargspec.annotations.get(
+            self._fullargspec.args[i])
+        if arg_annotation == core_tf_types.TensorLike:
+          args[i] = ops.convert_to_tensor(arg)
+      else:
```

Comment:
This isn't immediately obvious - please add a note explaining it (or alink to the docs of inspect.getfullargspec if it's available.

Reply:
The docs are available so I'll link it.